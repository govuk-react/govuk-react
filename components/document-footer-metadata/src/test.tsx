import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import { DocumentFooterMetadata } from '.';

const example = ['example'];
const multiExample = ['example', 'exampleexample'];
const otherExample = [
  {
    id: 0,
    title: 'example',
    content: ['example'],
  },
];
const wrapper = <DocumentFooterMetadata from={example} />;
const wrapperMultiple = <DocumentFooterMetadata partOf={multiExample} />;
const wrapperOther = <DocumentFooterMetadata other={otherExample} />;

describe('DocumentFooterMetadata', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DocumentFooterMetadata>Example</DocumentFooterMetadata>, div);
  });

  it('should render a from item', () => {
    const output = mount(wrapper);
    expect(output.find('ul')).toHaveLength(1);
  });

  it('should render multiple partof items', () => {
    expect(mount(wrapperMultiple).find('ul li')).toHaveLength(2);
  });

  it('should render a other items', () => {
    expect(mount(wrapperOther).find('ul li')).toHaveLength(1);
  });

  it('matches snapshot', () => {
    expect(mount(wrapperMultiple)).toMatchSnapshot('enzyme.mount');
  });
});
