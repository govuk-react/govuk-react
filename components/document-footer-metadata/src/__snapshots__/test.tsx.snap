// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentFooterMetadata matches snapshot: enzyme.mount 1`] = `
.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-top: 0;
  margin-bottom: 15px;
  list-style-type: none;
  padding-left: 0;
}

.c2 .c2 {
  margin-top: 10px;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
}

.c4 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  margin-bottom: 0;
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c3 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    margin-bottom: 20px;
  }
}

<DocumentFooterMetadata
  partOf={
    Array [
      "example",
      "exampleexample",
    ]
  }
>
  <src__StyledContainer>
    <div
      className="c0"
    >
      <src__StyledContainer>
        <div
          className="c0"
        />
      </src__StyledContainer>
      <src__StyledContainer>
        <div
          className="c0"
        >
          <div>
            <src__StyledParagraph>
              <p
                className="c1"
              >
                Part of:
              </p>
            </src__StyledParagraph>
            <UnorderedList
              listStyleType="none"
            >
              <OrderedList
                as="ul"
                listStyleType="none"
              >
                <ul
                  className="c2 c3"
                >
                  <src__StyledDefinition
                    key=".0"
                  >
                    <li
                      className="c4"
                    >
                      example
                    </li>
                  </src__StyledDefinition>
                  <src__StyledDefinition
                    key=".1"
                  >
                    <li
                      className="c4"
                    >
                      exampleexample
                    </li>
                  </src__StyledDefinition>
                </ul>
              </OrderedList>
            </UnorderedList>
          </div>
        </div>
      </src__StyledContainer>
      <src__StyledContainer>
        <div
          className="c0"
        />
      </src__StyledContainer>
    </div>
  </src__StyledContainer>
</DocumentFooterMetadata>
`;
